File: solutions/0073-set-matrix-zeroes/solution.java
class Solution {
    public void setZeroes(int[][] matrix) {
        int rlen= matrix.length;
        int rcol = matrix[0].length;

        boolean[] row = new boolean[rlen];    
        boolean[] col = new boolean[rcol];    

        for(int i=0;i< rlen ; i++){

            for(int j=0;j<rcol;j++){
                if(matrix[i][j] == 0){
                    row[i] = true;
                    col[j] = true;
                }
            }
        }

         for(int i=0;i< rlen ; i++){
            for(int j=0;j<rcol;j++){
                if(row[i] || col[j]){
                    matrix[i][j] =0;
                }
            }
        }
    }
}

File: solutions/0073-set-matrix-zeroes/solution.py
class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.

        """
        r_len = len(matrix)
        c_len = len(matrix[0])
        row = [0]*r_len
        col = [0]*c_len
        for i in range(r_len):
            for j in range(c_len):
                if matrix[i][j] == 0:
                    row[i] = 1
                    col[j] = 1
            
        for i in range(r_len):
            for j in range(c_len):
                if row[i] == 1 or col[j] == 1:
                    matrix[i][j]=0

                
